### PR TITLE
Improve error log in k8s discovery

### PR DESCRIPTION
**Pull request by vinicius-colutti**
Add ServiceAccount in KubernetesClient to show the specific serviceAccount and namespace, for Forbidden error

I am creating this PR, to solve this issue, all details are written in the issue

Fixes https://github.com/hazelcast/hazelcast/issues/24706

Backport of: INSERT_LINK_TO_THE_ORIGINAL_PR_HERE

EE PR: INSERT_LINK_TO_THE_EE_PR_HERE

Breaking changes (list specific methods/types/messages):
* API
* client protocol format
* serialized form
* snapshot format

Checklist:
- [ ] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [ ] Label `Add to Release Notes` or `Not Release Notes content` set
- [ ] Request reviewers if possible
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc

### CODE REVIEW

**Comment by cagric0** [comment]
Hi @vinicius-colutti, Thanks for your effort and the PR.

As you mentioned, sometimes the logs are not enough to understand the issue. For just printing serviceAccount, we are not planning to take it in the client config. The user is already knew which serviceAccount is used in which namespace(created by himself for the client). But as you suggest, printing additional logs would be helpful to understand the issue(missing permission in serviceAccount). 

Here is the updated version I am thinking:
```
LOGGER.warning("Kubernetes API access is forbidden! Starting standalone. To use Hazelcast Kubernetes discovery,"
 + " configure the required RBAC. For 'default' service account in 'default' namespace execute: "
 + "`kubectl apply -f https://raw.githubusercontent.com/hazelcast/hazelcast/master/kubernetes-rbac.yaml`" 
 + "If you want to use different service account and different namespace, you can update the mentioned rbac.yaml file accordingly and use it");

LOGGER.warning("Error Kubernetes API Cause: ", e.getCause());
LOGGER.warning("Error Kubernetes API Cause message: "+ e.getCause().getMessage());
```

**Comment by vinicius-colutti** [comment]
Hi @cagric0 Thanks for your review, I added the serviceAccount in the clientConfig, because I have a specific situation and sometimes users don't recognize their serviceAccount or namespace, but I know and understand now that this situation is not common.

So I agree with your suggestion and changed this PR with these changes, thank you!!

**Comment by kwart** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/kubernetes/KubernetesApiEndpointResolver.java [hunk 1]
```diff
@@ -190,4 +190,4 @@ void start() {
     void destroy() {
         client.destroy();
     }
-}
+}
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/kubernetes/KubernetesApiEndpointResolver.java:193]: Why do you change the file end here?
This should be reverted - including other places with the same change.

File: hazelcast/src/main/java/com/hazelcast/kubernetes/KubernetesClient.java [hunk 2]
```diff
@@ -617,8 +617,12 @@ private List<Endpoint> handleKnownException(RestClientException e) {
         } else if (e.getHttpErrorCode() == HTTP_FORBIDDEN) {
             if (!isKnownExceptionAlreadyLogged) {
                 LOGGER.warning("Kubernetes API access is forbidden! Starting standalone. To use Hazelcast Kubernetes discovery,"
-                        + " configure the required RBAC. For 'default' service account in 'default' namespace execute: "
-                        + "`kubectl apply -f https://raw.githubusercontent.com/hazelcast/hazelcast/master/kubernetes-rbac.yaml`");
+				 + " configure the required RBAC. For 'default' service account in 'default' namespace execute: "
+				 + "`kubectl apply -f https://raw.githubusercontent.com/hazelcast/hazelcast/master/kubernetes-rbac.yaml`" 
+				 + "If you want to use different service account and different namespace, you can update the mentioned rbac.yaml file accordingly and use it");
+				LOGGER.warning("Error Kubernetes API Cause: ", e.getCause());
+				LOGGER.warning("Error Kubernetes API Cause message: "+ e.getCause().getMessage());
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/kubernetes/KubernetesClient.java:624]: The formatting doesn't seem to be correct here. Moreover, it doesn't make sense to split the log into 3 messages. Could you use just the first one, but add the `e` as last parameter to the `warning(..., e);` method?

**Review by kwart — CHANGES_REQUESTED** [review]
Thank you for adding more details with solution description to the log message. It's a nice improvement. I suggest to use just the single log message and revert unrelated formatting changes.

**Review by vinicius-colutti — COMMENTED** [review]
Hi @kwart  Thanks for your review.

Can you check again?

**Comment by kwart** [comment]
run-lab-run

**Review by kwart — APPROVED** [review]

**Comment by vinicius-colutti** [comment]
run-lab-run

**Comment by kwart** [comment]
run-lab-run

**Review by cagric0 — APPROVED** [review]

**Comment by kwart** [comment]
Merged. Thanks for the fix, @vinicius-colutti !

**Pull request merged**